### PR TITLE
[WIP] std.path: -dip1000 compilable

### PR DESCRIPTION
**Pull request by carblue**
https://github.com/dlang/phobos/blob/master/dip1000.mak with aa[std.path]=-dip1000
Errors when running: make -f posix.mak std/path.test
```
...
T=`mktemp -d /tmp/.dmd-run-test.XXXXXX` &&                                                              \
  (                                                                                                     \
    ../dmd/generated/linux/release/64/dmd -od$T -conf= -I../druntime/import  -w -de -dip25 -m64 -fPIC -transition=complex -O -release -dip1000  -main -unittest generated/linux/release/64/libphobos2.a -defaultlib= -debuglib= -L-ldl -cov -run std/path.d ;     \
    RET=$? ; rm -rf $T ; exit $RET                                                                   \
  )
std/path.d(1273): Error: @safe function std.path.__unittest_L1269_C7 cannot call @system function std.path.testAliasedString!(withExtension, string).testAliasedString
std/path.d(1389): Error: @safe function std.path.__unittest_L1385_C7 cannot call @system function std.path.testAliasedString!(withDefaultExtension, string).testAliasedString
std/path.d(2881): Error: @safe function std.path.__unittest_L2879_C7 cannot call @system function std.path.relativePath!cast(CaseSensitive)true.relativePath
std/path.d(2885): Error: @safe function std.path.__unittest_L2879_C7 cannot call @system function std.path.relativePath!cast(CaseSensitive)true.relativePath
std/path.d(2886): Error: @safe function std.path.__unittest_L2879_C7 cannot call @system function std.path.relativePath!cast(CaseSensitive)true.relativePath
std/path.d(2887): Error: @safe function std.path.__unittest_L2879_C7 cannot call @system function std.path.relativePath!cast(CaseSensitive)true.relativePath
std/path.d(2888): Error: @safe function std.path.__unittest_L2879_C7 cannot call @system function std.path.relativePath!cast(CaseSensitive)true.relativePath
std/path.d(2889): Error: @safe function std.path.__unittest_L2879_C7 cannot call @system function std.path.relativePath!cast(CaseSensitive)true.relativePath
std/path.d(2905): Error: @safe function std.path.__unittest_L2902_C7 cannot call @system function std.path.relativePath!cast(CaseSensitive)true.relativePath
std/path.d(2908): Error: @safe function std.path.__unittest_L2902_C7 cannot call @system function std.path.relativePath!cast(CaseSensitive)true.relativePath
std/path.d(2909): Error: @safe function std.path.__unittest_L2902_C7 cannot call @system function std.path.relativePath!cast(CaseSensitive)true.relativePath
std/path.d(2910): Error: @safe function std.path.__unittest_L2902_C7 cannot call @system function std.path.relativePath!cast(CaseSensitive)true.relativePath
```
Still looking for a fix of errors (behind std.path.testAliasedString)  with std.range.package. chain.Result

### CODE REVIEW

File: std/range/package.d [hunk 1]
```diff
@@ -1415,7 +1415,7 @@ private struct ChooseResult(R1, R2)
     private bool r1Chosen;
 
     private static auto ref actOnChosen(alias foo, ExtraArgs ...)(ref ChooseResult r,
-            auto ref ExtraArgs extraArgs)
+            auto ref ExtraArgs extraArgs) @trusted
```
Comment (wilzbach) [inline std/range/package.d]: We can't just set a function that takes an arbitrary function as `@trusted`

File: std/path.d [hunk 2]
```diff
@@ -1270,7 +1270,7 @@ private auto _withExtension(R, C)(R path, C[] ext)
 {
     import std.algorithm.comparison : equal;
 
-    assert(testAliasedString!withExtension("file", "ext"));
+//    assert(testAliasedString!withExtension("file", "ext"));
```
Comment (wilzbach) [inline std/path.d]: Move to a `version(DIP1000) {} else` unittest?

File: std/range/package.d (hunk 1 above)
Comment (carblue) [reply std/range/package.d]: Yes, I'm aware; I'm thinking about a way how to cope with that

File: std/path.d (hunk 2 above)
Comment (carblue) [reply std/path.d]: Yes, if I don't find a general solution

Comment (wilzbach) [reply std/path.d]: Looks like you didn't, so I suggest to go with `version(DIP1000) {} else` for now, s.t. we finally get entire Phobos working and then trim the small blocks out?

**Comment by wilzbach** [comment]
Resolved as Phobos now compiles with -dip1000

**Pull request closed without merging**